internal/git/head.go

```diff
@@ -16,46 +24,42 @@ func HeadTagOrCommit(repoPath string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("unable fetch head: %w", err)
 	}
+
 	tagRefs, _ := r.Tags()
 	var tagName string
 
 	_ = tagRefs.ForEach(func(t *plumbing.Reference) error {
 		if t.Hash().String() == ref.Hash().String() {
+			// for lightweight tags
 			tagName = t.Name().Short()
 			return fmt.Errorf("found")
 		}
-		return nil
-	})
 
-	if tagName != "" {
-		return tagName, nil
-	}
-
-	return ref.Hash().String(), nil
-}
+		// a little extra work for annotated tags
```

Comment:
I don't understand this comment.

Reply:
I'm not opposed to having a comment; it might be nice to just say something like, "annotated tags have their own object in the commit graph and need to be dereferenced" or something; I just don't think the comment as is makes sense to someone who doesn't have context on this PR.